browser/ui/tabs/test/tree_tab_node_unittest.cc

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/tabs/public/tree_tab_node.h"
+
+#include "chrome/browser/ui/tabs/tab_group_desktop.h"
+#include "chrome/test/base/testing_profile.h"
+#include "components/tabs/public/mock_tab_interface.h"
+#include "components/tabs/public/pinned_tab_collection.h"
+#include "components/tabs/public/split_tab_collection.h"
+#include "components/tabs/public/split_tab_visual_data.h"
+#include "components/tabs/public/tab_group_tab_collection.h"
+#include "components/tabs/public/unpinned_tab_collection.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+class TreeTabNodeUnitTest : public testing::Test {
+ protected:
+  TestingProfile* profile() { return testing_profile_.get(); }
+
+ private:
```

Comment:
nit: I believe it's fine to have test class variables in `protected:` area to not end up with a bunch of getters like `TestingProfile* profile() { return testing_profile_.get(); }`.

having `private:` makes sense for "Base" test classes that are later reused.

Reply:
I think the advantage of the accessor here is that it tells you none of the tests are changing the value of `testing_profile_`.